### PR TITLE
feature/multiprocess_parser doc: custom_tag

### PR DESCRIPTION
**Pull request by fabianherrera**
doc: module doc, type and specs for custom tag funcionality. (combinators and translators) 
due PR 84 comments.

### CODE REVIEW

File: lib/liquid/combinators/tags/custom_tag.ex [hunk 1]
```diff
@@ -1,7 +1,27 @@
 defmodule Liquid.Combinators.Tags.CustomTag do
+  @moduledoc """
+  Implementation of custom tag. "Tags" are tags that take any number of arguments, but do not contain a block of template code.
+  To create a new tag, Use Liquid.Register module and register your tag with Liquid.Register.register/3.
+  The register tag  takes three arguments: the user-facing name of the tag, the module where code of parsing/rendering is located
+  and the type that implements it (tag or block).
+
+  ```
+    {% MyCustomTag argument1 =1, argument2, argument3 = 5 %}
```
Comment (Spadavecchia) [inline lib/liquid/combinators/tags/custom_tag.ex:9]: argument1 = 1

File: lib/liquid/combinators/tags/custom_tag.ex [hunk 2]
```diff
@@ -56,6 +70,10 @@ defmodule Liquid.Combinators.Tags.CustomTag do
     |> traverse({Liquid.Combinators.Tags.CustomTag, :check_string, []})
   end
 
+  @doc """
+  Parses a `Liquid` Custom tag's name, isolates custom tag name from markup.
```
Comment (Spadavecchia) [inline lib/liquid/combinators/tags/custom_tag.ex:74]: Different function, same documentation?

File: lib/liquid/combinators/tags/custom_tag.ex [hunk 3]
```diff
@@ -84,13 +102,16 @@ defmodule Liquid.Combinators.Tags.CustomTag do
...
-  def all_tags do
+  defp all_tags do
     List.flatten([
       Liquid.Combinators.Tags.CustomTag.end_register_tag_name()
       | Liquid.Combinators.Tags.CustomTag.liquid_tags()
     ])
   end
-
+  @doc """
+  Return a List of Liquid reserved tag's name, includind end block types.
```
Comment (Spadavecchia) [inline lib/liquid/combinators/tags/custom_tag.ex:112]: Returns, including

File: lib/liquid/combinators/tags/custom_tag.ex [hunk 4]
```diff
@@ -120,7 +141,10 @@ defmodule Liquid.Combinators.Tags.CustomTag do
       ""
     ]
   end
-
+  @doc """
+  Return a List of Liquid reserved end tag's name, includind tag names.
```
Comment (Spadavecchia) [inline lib/liquid/combinators/tags/custom_tag.ex:145]: Returns, including. Check the rest Returns, Translates, etc

File: lib/liquid/combinators/tags/custom_tag.ex (hunk 1 above)
Comment (fabianherrera) [reply lib/liquid/combinators/tags/custom_tag.ex:9]: Its possible.  Done

File: lib/liquid/combinators/tags/custom_tag.ex (hunk 2 above)
Comment (fabianherrera) [reply lib/liquid/combinators/tags/custom_tag.ex:74]: Right. It was changed.

File: lib/liquid/combinators/tags/custom_tag.ex (hunk 3 above)
Comment (fabianherrera) [reply lib/liquid/combinators/tags/custom_tag.ex:112]: Done.

File: lib/liquid/combinators/tags/custom_tag.ex (hunk 4 above)
Comment (fabianherrera) [reply lib/liquid/combinators/tags/custom_tag.ex:145]: Done.

**Pull request closed without merging**